space_packet_parser/xtcedef.py

```diff
@@ -1220,20 +1275,29 @@ def __init__(self, encoding: str = 'utf-8',
             Function that linearly adjusts a size. e.g. if the size reference parameter gives a length in bytes, the
             linear adjuster should multiply by 8 to give the size in bits.
         """
-        if encoding not in ['utf-8', 'utf-16-le', 'utf-16-be']:
-            raise ValueError(
-                f"Got encoding={encoding}. Encoding must be one of utf-8, utf-16-le, or utf-16-be (note that"
-                f"endianness must be specified for utf-16 encoding.")
+        if encoding.upper() not in self._supported_encodings:
```

Comment:
This will fail for the `Windows-1252` comparison because it isn't uppercased in the `_supported_encodings`.

The flipping between incoming/upper/lower is a bit confusing through here. Can we not do the upper/lower conversions and rely on the user matching the spec? Or store this as upper-case on the `self.encoding` attribute and then special-case Windows later in the lookup.

![image](https://github.com/user-attachments/assets/42feb44d-9eff-4732-9e55-557c0247e8a3)

Reply:
I just went down a rabbit hole of defining canonicalized enums for all these string parameters, allowing aliases but coercing to standard representations, and just decided it's more complicated than it's worth. I'm OK with demanding strict adherence to the spec.

How do you feel about this when it comes to integer encodings? For example `signed` is not a valid encoding according to XTCE. It must be `twosComplement`.